pints/plot.py

```diff
@@ -8,6 +8,201 @@
 #
 from __future__ import division
 
+def log_likelihood_1d(log_likelihood, parameters, boundaries=None, n_eva=20, chain=None):
+    """
+    Takes a set of parameters as input and creates and returns a plot showing 
+    the log-likelihood function of each parameter with all the other 
+    parameters held at the given set of parameters.
+    
+    Arguments:
+
+    `log_likelihood`
+        A class (or that extends from) :class:`pints.LogLikelihood` of the 
+        problem of interest.
+    `parameters`
+        A set of parameters, which the `log_likelihood` will be evaluated at, 
+        with the same dimension as the `log_likelihood` will accept.
+    `boundaries`
+        A class (or that extends from) :class:`pints.Boundaries` which defines 
+        the boundaries of each parameters.  
+    `n_eva`
+        (Optional) The number of evaluation of each parameter.
+    `chain`
+        (Optional) A markov chain of shape `(samples, dimension)`, where 
+        `samples` is the number of samples in the chain and `dimension` is the 
+        number of parameters. This will be used to define the boundaries. Only 
+        provide either `chain` or `boundaries`, but not both.
+
+    Returns a `matplotlib` figure object and axes handle.
+    """
+    import pints
+    import numpy as np
+    try:
+        import matplotlib.pyplot as plt
+    except:
+        raise ImportError("The method pints.plot.trace requires matplotlib")
+
+    n_param = len(parameters)
+    # Check dimension
+    if n_param != log_likelihood.dimension():
+        raise ValueError('Input parameters dimension must match the dimension'
+                        ' of the log_likelihood accept')
+    # Check input
+    if not isinstance(boundaries, pints.Boundaries) and boundaries is not None:
+        raise TypeError('The argument boundaries must be a type of or extended'
+                ' from pints.Boundaries')
+    if chain is None and boundaries is not None:
+        def_bound = False
+    elif chain is not None and boundaries is None:
+        def_bound = True
+    else:
+        raise TypeError('One of the input chain or boundaries (and only one)'
+                        ' is expected')
+
+    fig, axes = plt.subplots(n_param, 1, figsize=(6, 2*n_param))
```

Comment:
I'm worried about all the fixed plot sizes in `plot.py`. We're choosing them just because they look good in our examples